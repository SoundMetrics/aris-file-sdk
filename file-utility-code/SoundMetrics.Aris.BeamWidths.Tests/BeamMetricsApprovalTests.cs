using ApprovalTests;
using ApprovalTests.Namers;
using ApprovalTests.Reporters;
using Microsoft.VisualStudio.TestTools.UnitTesting;
using SoundMetrics.Aris.Headers;
using System.Text;

namespace SoundMetrics.Aris.BeamWidths.Tests
{
    [TestClass]
    [UseApprovalSubdirectory("approval-files"), UseReporter(typeof(DiffReporter))]
    public sealed class BeamMetricsApprovalTests
    {
        [TestMethod]
        public void BeamWidths_ARIS3000_128()
            => RunTest(new BeamConfig(ArisSystemType.Aris3000, 128, LensType.None));

        [TestMethod]
        public void BeamWidths_ARIS3000_64()
            => RunTest(new BeamConfig(ArisSystemType.Aris3000, 64, LensType.None));

        [TestMethod]
        public void BeamWidths_ARIS1200_48()
            => RunTest(new BeamConfig(ArisSystemType.Aris1200, 48, LensType.None));

        [TestMethod]
        public void BeamWidths_ARIS1800_48()
            => RunTest(new BeamConfig(ArisSystemType.Aris1800, 48, LensType.None));

        [TestMethod]
        public void BeamWidths_ARIS1800_96()
            => RunTest(new BeamConfig(ArisSystemType.Aris1800, 96, LensType.None));

        [TestMethod]
        public void BeamWidths_ARIS1200_Telephoto_48()
            => RunTest(new BeamConfig(ArisSystemType.Aris1200, 48, LensType.Telephoto));

        [TestMethod]
        public void BeamWidths_ARIS1800_Telephoto_48()
            => RunTest(new BeamConfig(ArisSystemType.Aris1200, 48, LensType.Telephoto));

        [TestMethod]
        public void BeamWidths_ARIS1800_Telephoto_96()
            => RunTest(new BeamConfig(ArisSystemType.Aris1800, 96, LensType.Telephoto));

        private static void RunTest(BeamConfig cfg)
        {
            var stringOutput = BuildStringOutput(cfg);
            Approvals.Verify(stringOutput);
        }

        private static string BuildStringOutput(BeamConfig cfg)
        {
            var beamInfos = Metrics.GetBeamInformation(cfg);
            var stringOutput = BuildOutput(beamInfos);
            return stringOutput;

            static string BuildOutput(BeamInfo[] beamInfos)
            {
                var buffer = new StringBuilder();
                buffer.AppendLine($"# of BeamInfo members: [{beamInfos.Length}]");

                foreach (var beamInfo in beamInfos)
                {
                    buffer.AppendLine(beamInfo.ToString());
                }

                return buffer.ToString();
            }
        }
    }
}
