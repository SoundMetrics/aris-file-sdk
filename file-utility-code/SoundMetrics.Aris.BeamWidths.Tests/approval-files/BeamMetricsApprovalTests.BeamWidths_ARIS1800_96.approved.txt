# of BeamInfo members: [96]
BeamInfo { BeamNumber = 0, Center = -13.749, Left = -13.8562, Right = -13.6419 }
BeamInfo { BeamNumber = 1, Center = -13.5347, Left = -13.6419, Right = -13.4261 }
BeamInfo { BeamNumber = 2, Center = -13.3174, Left = -13.4261, Right = -13.2075 }
BeamInfo { BeamNumber = 3, Center = -13.0975, Left = -13.2075, Right = -12.9867 }
BeamInfo { BeamNumber = 4, Center = -12.8758, Left = -12.9867, Right = -12.7628 }
BeamInfo { BeamNumber = 5, Center = -12.6498, Left = -12.7628, Right = -12.5348 }
BeamInfo { BeamNumber = 6, Center = -12.4197, Left = -12.5348, Right = -12.3026 }
BeamInfo { BeamNumber = 7, Center = -12.1855, Left = -12.3026, Right = -12.0666 }
BeamInfo { BeamNumber = 8, Center = -11.9476, Left = -12.0666, Right = -11.8273 }
BeamInfo { BeamNumber = 9, Center = -11.7069, Left = -11.8273, Right = -11.5853 }
BeamInfo { BeamNumber = 10, Center = -11.4636, Left = -11.5853, Right = -11.3405 }
BeamInfo { BeamNumber = 11, Center = -11.2173, Left = -11.3405, Right = -11.0925 }
BeamInfo { BeamNumber = 12, Center = -10.9676, Left = -11.0925, Right = -10.841 }
BeamInfo { BeamNumber = 13, Center = -10.7144, Left = -10.841, Right = -10.5862 }
BeamInfo { BeamNumber = 14, Center = -10.4579, Left = -10.5862, Right = -10.3282 }
BeamInfo { BeamNumber = 15, Center = -10.1985, Left = -10.3282, Right = -10.0669 }
BeamInfo { BeamNumber = 16, Center = -9.9352, Left = -10.0669, Right = -9.8016 }
BeamInfo { BeamNumber = 17, Center = -9.668, Left = -9.8016, Right = -9.5321 }
BeamInfo { BeamNumber = 18, Center = -9.3961, Left = -9.5321, Right = -9.2578 }
BeamInfo { BeamNumber = 19, Center = -9.1195, Left = -9.2578, Right = -8.979 }
BeamInfo { BeamNumber = 20, Center = -8.8385, Left = -8.979, Right = -8.696 }
BeamInfo { BeamNumber = 21, Center = -8.5535, Left = -8.696, Right = -8.4095 }
BeamInfo { BeamNumber = 22, Center = -8.2654, Left = -8.4095, Right = -8.1196 }
BeamInfo { BeamNumber = 23, Center = -7.9738, Left = -8.1196, Right = -7.826 }
BeamInfo { BeamNumber = 24, Center = -7.6781, Left = -7.826, Right = -7.5287 }
BeamInfo { BeamNumber = 25, Center = -7.3793, Left = -7.5287, Right = -7.2284 }
BeamInfo { BeamNumber = 26, Center = -7.0774, Left = -7.2284, Right = -6.9251 }
BeamInfo { BeamNumber = 27, Center = -6.7727, Left = -6.9251, Right = -6.619 }
BeamInfo { BeamNumber = 28, Center = -6.4652, Left = -6.619, Right = -6.3101 }
BeamInfo { BeamNumber = 29, Center = -6.1549, Left = -6.3101, Right = -5.9983 }
BeamInfo { BeamNumber = 30, Center = -5.8416, Left = -5.9983, Right = -5.6831 }
BeamInfo { BeamNumber = 31, Center = -5.5246, Left = -5.6831, Right = -5.3646 }
BeamInfo { BeamNumber = 32, Center = -5.2046, Left = -5.3646, Right = -5.0435 }
BeamInfo { BeamNumber = 33, Center = -4.8824, Left = -5.0435, Right = -4.7201 }
BeamInfo { BeamNumber = 34, Center = -4.5578, Left = -4.7201, Right = -4.3944 }
BeamInfo { BeamNumber = 35, Center = -4.231, Left = -4.3944, Right = -4.0665 }
BeamInfo { BeamNumber = 36, Center = -3.9019, Left = -4.0665, Right = -3.7365 }
BeamInfo { BeamNumber = 37, Center = -3.5711, Left = -3.7365, Right = -3.4049 }
BeamInfo { BeamNumber = 38, Center = -3.2387, Left = -3.4049, Right = -3.0716 }
BeamInfo { BeamNumber = 39, Center = -2.9044, Left = -3.0716, Right = -2.7365 }
BeamInfo { BeamNumber = 40, Center = -2.5686, Left = -2.7365, Right = -2.4 }
BeamInfo { BeamNumber = 41, Center = -2.2314, Left = -2.4, Right = -2.062 }
BeamInfo { BeamNumber = 42, Center = -1.8925, Left = -2.062, Right = -1.7222 }
BeamInfo { BeamNumber = 43, Center = -1.5519, Left = -1.7222, Right = -1.3809 }
BeamInfo { BeamNumber = 44, Center = -1.2098, Left = -1.3809, Right = -1.0381 }
BeamInfo { BeamNumber = 45, Center = -0.8664, Left = -1.0381, Right = -0.694 }
BeamInfo { BeamNumber = 46, Center = -0.5216, Left = -0.694, Right = -0.3487 }
BeamInfo { BeamNumber = 47, Center = -0.1757, Left = -0.3487, Right = -0.0026 }
BeamInfo { BeamNumber = 48, Center = 0.1706, Left = -0.0026, Right = 0.3438 }
BeamInfo { BeamNumber = 49, Center = 0.5169, Left = 0.3438, Right = 0.6899 }
BeamInfo { BeamNumber = 50, Center = 0.8628, Left = 0.6899, Right = 1.0352 }
BeamInfo { BeamNumber = 51, Center = 1.2076, Left = 1.0352, Right = 1.3793 }
BeamInfo { BeamNumber = 52, Center = 1.551, Left = 1.3793, Right = 1.7221 }
BeamInfo { BeamNumber = 53, Center = 1.8931, Left = 1.7221, Right = 2.0634 }
BeamInfo { BeamNumber = 54, Center = 2.2337, Left = 2.0634, Right = 2.4032 }
BeamInfo { BeamNumber = 55, Center = 2.5726, Left = 2.4032, Right = 2.7412 }
BeamInfo { BeamNumber = 56, Center = 2.9098, Left = 2.7412, Right = 3.0777 }
BeamInfo { BeamNumber = 57, Center = 3.2456, Left = 3.0777, Right = 3.4128 }
BeamInfo { BeamNumber = 58, Center = 3.5799, Left = 3.4128, Right = 3.7461 }
BeamInfo { BeamNumber = 59, Center = 3.9123, Left = 3.7461, Right = 4.0777 }
BeamInfo { BeamNumber = 60, Center = 4.2431, Left = 4.0777, Right = 4.4077 }
BeamInfo { BeamNumber = 61, Center = 4.5722, Left = 4.4077, Right = 4.7356 }
BeamInfo { BeamNumber = 62, Center = 4.899, Left = 4.7356, Right = 5.0613 }
BeamInfo { BeamNumber = 63, Center = 5.2236, Left = 5.0613, Right = 5.3847 }
BeamInfo { BeamNumber = 64, Center = 5.5458, Left = 5.3847, Right = 5.7058 }
BeamInfo { BeamNumber = 65, Center = 5.8658, Left = 5.7058, Right = 6.0243 }
BeamInfo { BeamNumber = 66, Center = 6.1828, Left = 6.0243, Right = 6.3395 }
BeamInfo { BeamNumber = 67, Center = 6.4961, Left = 6.3395, Right = 6.6513 }
BeamInfo { BeamNumber = 68, Center = 6.8064, Left = 6.6513, Right = 6.9602 }
BeamInfo { BeamNumber = 69, Center = 7.1139, Left = 6.9602, Right = 7.2663 }
BeamInfo { BeamNumber = 70, Center = 7.4186, Left = 7.2663, Right = 7.5696 }
BeamInfo { BeamNumber = 71, Center = 7.7205, Left = 7.5696, Right = 7.8699 }
BeamInfo { BeamNumber = 72, Center = 8.0193, Left = 7.8699, Right = 8.1672 }
BeamInfo { BeamNumber = 73, Center = 8.315, Left = 8.1672, Right = 8.4608 }
BeamInfo { BeamNumber = 74, Center = 8.6066, Left = 8.4608, Right = 8.7507 }
BeamInfo { BeamNumber = 75, Center = 8.8947, Left = 8.7507, Right = 9.0372 }
BeamInfo { BeamNumber = 76, Center = 9.1797, Left = 9.0372, Right = 9.3202 }
BeamInfo { BeamNumber = 77, Center = 9.4607, Left = 9.3202, Right = 9.599 }
BeamInfo { BeamNumber = 78, Center = 9.7373, Left = 9.599, Right = 9.8733 }
BeamInfo { BeamNumber = 79, Center = 10.0092, Left = 9.8733, Right = 10.1428 }
BeamInfo { BeamNumber = 80, Center = 10.2764, Left = 10.1428, Right = 10.4081 }
BeamInfo { BeamNumber = 81, Center = 10.5397, Left = 10.4081, Right = 10.6694 }
BeamInfo { BeamNumber = 82, Center = 10.7991, Left = 10.6694, Right = 10.9274 }
BeamInfo { BeamNumber = 83, Center = 11.0556, Left = 10.9274, Right = 11.1822 }
BeamInfo { BeamNumber = 84, Center = 11.3088, Left = 11.1822, Right = 11.4337 }
BeamInfo { BeamNumber = 85, Center = 11.5585, Left = 11.4337, Right = 11.6817 }
BeamInfo { BeamNumber = 86, Center = 11.8048, Left = 11.6817, Right = 11.9265 }
BeamInfo { BeamNumber = 87, Center = 12.0481, Left = 11.9265, Right = 12.1685 }
BeamInfo { BeamNumber = 88, Center = 12.2888, Left = 12.1685, Right = 12.4078 }
BeamInfo { BeamNumber = 89, Center = 12.5267, Left = 12.4078, Right = 12.6438 }
BeamInfo { BeamNumber = 90, Center = 12.7609, Left = 12.6438, Right = 12.876 }
BeamInfo { BeamNumber = 91, Center = 12.991, Left = 12.876, Right = 13.104 }
BeamInfo { BeamNumber = 92, Center = 13.217, Left = 13.104, Right = 13.3279 }
BeamInfo { BeamNumber = 93, Center = 13.4387, Left = 13.3279, Right = 13.5487 }
BeamInfo { BeamNumber = 94, Center = 13.6586, Left = 13.5487, Right = 13.7673 }
BeamInfo { BeamNumber = 95, Center = 13.8759, Left = 13.7673, Right = 13.9831 }
